workflow/snakemake_rules/upload.smk

```diff
@@ -55,25 +64,31 @@ def compute_files_to_upload(wildcards):
         files_to_upload["additional_info.tsv.zst"] =     f"data/{database}/additional_info.tsv"
         files_to_upload["flagged_metadata.txt.zst"] =    f"data/{database}/flagged_metadata.txt"
 
-    nextclade_sequences_path = checkpoints.get_sequences_without_nextclade_annotations.get().output.fasta
-    if os.path.getsize(nextclade_sequences_path) > 0:
-        files_to_upload["nextclade.tsv.gz"] =           f"data/{database}/nextclade.tsv"
-        files_to_upload["aligned.fasta.xz"] =           f"data/{database}/aligned.fasta"
-
-        files_to_upload["nextclade.tsv.zst"] =           f"data/{database}/nextclade.tsv"
-        files_to_upload["aligned.fasta.zst"] =           f"data/{database}/aligned.fasta"
     return files_to_upload
 
+def files_to_upload(wildcards):
+    files_to_upload_dict = compute_files_to_upload(wildcards)
+    return [f"data/{database}/{remote_file}.upload" for remote_file in files_to_upload_dict.keys()]
 
-rule upload:
-    input:
-        unpack(compute_files_to_upload)
+
+rule upload_single:
```

Comment:
We introduce a rule `upload_single` that uploads a single file as soon as it's done, rather than all files sequentially and in the end.